src/NuGet/Microsoft.NETCore.Compilers/Microsoft.NETCore.Compilers.Package.csproj

```diff
@@ -15,21 +15,22 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\..\Compilers\CSharp\csc\csc.csproj" PrivateAssets="all"/>
-    <ProjectReference Include="..\..\Compilers\VisualBasic\vbc\vbc.csproj" PrivateAssets="all"/>
-    <ProjectReference Include="..\..\Interactive\csi\csi.csproj" PrivateAssets="all"/>
-    <ProjectReference Include="..\..\Compilers\Core\MSBuildTask\Microsoft.Build.Tasks.CodeAnalysis.csproj" PrivateAssets="all"/>
-    <ProjectReference Include="..\..\Compilers\Server\VBCSCompiler\VBCSCompiler.csproj" PrivateAssets="all"/>
+    <ProjectReference Include="..\..\Compilers\CSharp\csc\csc.csproj"/>
+    <ProjectReference Include="..\..\Compilers\VisualBasic\vbc\vbc.csproj"/>
+    <ProjectReference Include="..\..\Interactive\csi\csi.csproj"/>
+    <ProjectReference Include="..\..\Compilers\Core\MSBuildTask\Microsoft.Build.Tasks.CodeAnalysis.csproj"/>
+    <ProjectReference Include="..\..\Compilers\Server\VBCSCompiler\VBCSCompiler.csproj"/>
   </ItemGroup>
 
   <ItemGroup>
-    <_ExplicitProjectReference Include="@(ProjectReference)" />
+    <ProjectReference Update="@(ProjectReference)"
+                      Targets="Publish"
```

Comment:
Will this implicitly build the projects as well? Is there a way to skip that if desired? Probably don't want to do that here but curious none the less.

Reply:
Yes, it will build the projects. This basically calls `Publish` target on the referenced project. By default, `Publish` builds.

Reply:
There _is_ a way to disconnect publish from build, used by `dotnet publish --no-build`: set the property `NoBuild=true`. Details in https://github.com/dotnet/sdk/pull/2111 and https://github.com/dotnet/cli/issues/5331.

But you have to be careful about using that, since it creates a new set of global properties and could race with other builds of the same project. If you want to do publish-without-build in an `<MSBuild` task, you should do it separately, **not** by specifying metadata on the `ProjectReference` item.